userComponent/src/main/kotlin/com/gravatar/app/usercomponent/domain/usecase/FetchAvatarsUseCase.kt

```diff
@@ -0,0 +1,27 @@
+package com.gravatar.app.usercomponent.domain.usecase
+
+import com.gravatar.app.usercomponent.data.AvatarCacheBusterStorage
+import com.gravatar.app.usercomponent.domain.repository.UserRepository
+import com.gravatar.restapi.models.Avatar
+
+internal class FetchAvatarsUseCase(
+    private val userRepository: UserRepository,
+    private val avatarCacheBusterStorage: AvatarCacheBusterStorage,
+) : FetchUserAvatars {
+
+    override suspend fun invoke(isRefreshing: Boolean): Result<List<Avatar>> {
+        return userRepository.getAvatars()
+            .onSuccess { avatars ->
+                if (isRefreshing) {
+                    val avatarID = avatars.firstOrNull { it.selected == true }?.imageId
+                    avatarID?.let { id ->
+                        avatarCacheBusterStorage.saveAvatarCacheBuster(id)
+                    }
+                }
+            }
+    }
```

Comment:
I have copied the mechanism other requests use to update the avatar. I hope it's the correct one in this case also.